package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -537,254 +792,296 @@ def _update_water_selection(self):
+        this is not fool-proof as the atom type is not always a character but
+        the name pattern should catch most typical occurrences.
+
+        The distance from `atom` is calculated for all hydrogens in the residue
+        and only those within a cutoff are kept. The cutoff depends on the
+        heavy atom (more precisely, on its element, which is taken as the first
+        letter of its name ``atom.name[0]``) and is parameterized in
+        :attr:`HydrogenBondAnalysis.r_cov`. If no match is found then the
+        default of 1.5 Å is used.
 
-        Stores the water bridge data per frame as
-        :attr:`WaterBridgeAnalysis.timeseries` (see there for output
-        format).
 
         Parameters
         ----------
-        start : int (optional)
-            starting frame-index for analysis, ``None`` is the first one, 0.
-            `start` and `stop` are 0-based frame indices and are used to slice
-            the trajectory (if supported) [``None``]
-        stop : int (optional)
-            last trajectory frame for analysis, ``None`` is the last one
-            [``None``]
-        step : int (optional)
-            read every `step` between `start` (included) and `stop` (excluded),
-            ``None`` selects 1. [``None``]
-        verbose : bool (optional)
-             toggle progress meter output
-             :class:`~MDAnalysis.lib.log.ProgressMeter` [``True``]
-        debug : bool (optional)
-             enable detailed logging of debugging information; this can create
-             *very big* log files so it is disabled (``False``) by default;
-             setting `debug` toggles the debug status for
-             :class:`WaterBridgeAnalysis`, namely the value of
-             :attr:`WaterBridgeAnalysis.debug`.
+        atom : groups.Atom
+             heavy atom
 
-        See Also
-        --------
-        :meth:`WaterBridgeAnalysis.generate_table` :
-               processing the data into a different format.
+        Returns
+        -------
+        hydrogen_atoms : AtomGroup or []
+            list of hydrogens (can be a :class:`~MDAnalysis.core.groups.AtomGroup`)
+            or empty list ``[]`` if none were found.
         """
-        self._setup_frames(self.u.trajectory, start, stop, step)
+        try:
+            return atom.residue.atoms.select_atoms(
+                "(name H* 1H* 2H* 3H* or type H) and around {0:f} name {1!s}"
+                "".format(self.r_cov[atom.name[0]], atom.name))
+        except NoDataError:
+            return []
+
+    def logger_debug(self, *args):
+        if self.debug:
+            logger.debug(*args)
+
+
+    def _prepare(self):
+        # The distance for selection is defined as twice the maximum bond length of an O-H bond (2A)
+        # plus order of water bridge times the length of OH bond plus hydrogne bond distance
+        self.selection_distance = 1.0 * (2 * 2 + self.order * (2 + self.distance))
```

Comment:
Why `1.0 * `?

You should have `from __future__ import division` and any arrays should be declared with the proper float type when you make them. Such work-arounds to force float should not be necessary.

Reply:
Changed.